kore/src/Kore/Step/SMT/Translate.hs

```diff
@@ -297,17 +297,23 @@ translatePredicateWith translateTerm predicate =
               | builtinSort == Builtin.Bool.sort -> translateBool pat
               | builtinSort == Builtin.Int.sort -> translateInt pat
             _ -> case Cofree.tailF $ Recursive.project pat of
-                    VariableF _ -> do
-                        smtSort <- hoistMaybe $ translateSort sort
-                        translateUninterpreted smtSort pat
-                    ApplySymbolF app -> translateApplication app
+                    VariableF _ -> translateUninterpreted'
+                    ApplySymbolF app
+                      | isFunctionalPattern pat ->
+                          translateApplication app
+                          <|> translateUninterpreted'
```

Comment:
Why can't we add this to `translateApplication` itself?

Reply:
Thanks for suggesting this, I made the change and added some comments on the PR about the additional things I had to modify resulting from this.